File: lib/src/utils/Stack.dart
class Stack<T> {
    final List<T> _list = [];

    void push(T newItem) {
        _list.add(newItem);
    }

    T pop() {
        var result = _list.last;
        _list.removeLast();
        return result;
    }

    T? peek() {
        return _list.isNotEmpty ? _list.last : null;
    }
}

File: src/csharp/utils/Stack.cs
namespace O7;
using System.Collections.Generic;
using System.Linq;

public sealed class Stack<T> where T : class {
    private List<T> list;

    public Stack() {
        this.list = new List<T>();
    }

    public void push(T newItem) {
        list.Add(newItem);
    }

    public T pop() {
        var result = list[list.Count - 1];
        list.removeLast();
        return result;
    }

    public T peek() {
        return list.Any() ? list[list.Count - 1] : null;
    }
}
